pkg/kubectl/cmd/create_role.go

```diff
@@ -147,10 +166,17 @@ func (c *CreateRoleOptions) Validate(f cmdutil.Factory) error {
 	}
 
 	for _, r := range c.Resources {
-		_, err := mapper.ResourceFor(r)
-		if err != nil {
+		if len(r.Resource) == 0 {
+			return fmt.Errorf("resource must be specified if apiGroup/subresource specified")
+		}
+		if _, err := mapper.ResourceFor(schema.GroupVersionResource{Resource: r.Resource, Group: r.Group}); err != nil {
 			return err
 		}
+		if len(r.SubResource) > 0 {
+			if _, err := mapper.ResourceFor(schema.GroupVersionResource{Resource: r.SubResource}); err != nil {
```

Comment:
I don't see how this is validating what you think... without including the leading `<resource>/` bit, it can't check to see if this is actually a valid registered subresource, which means that this is probably a no-op